index.js

```diff
@@ -1,11 +1,29 @@
 var ask = function() {
   var formIFrame = document.createElement('iframe')
-  formIFrame.setAttribute(
-    'src',
-    'https://consent-form-www.devc.cloud.ogury.io/'
-  )
+  var onMessageReceived = function(event) {
+    if (
+      event.origin !== 'https://consent-form-www.devc.cloud.ogury.io' &&
+      event.origin !== 'https://consent-form-www.staging.cloud.ogury.io'
+    ) {
+      return
+    }
+    if (event.data.consent === undefined) {
+      return
+    }
+    var consentResult = JSON.parse(decodeURI(event.data.consent))
+    console.log('JSON CONSENT', consentResult)
+    return Promise.resolve(consentResult)
+  }
+  formIFrame.setAttribute('src', 'https://consent-form-www.devc.cloud.ogury.io')
   formIFrame.setAttribute('class', 'cmjs-form')
+  formIFrame.onbeforeunload = function() {
+    window.removeEventListener('message', onMessageReceived)
+  }
   document.body.appendChild(formIFrame)
+  formIFrame.onload = function() {
+    return window.addEventListener('message', onMessageReceived)
+  }
+  return Promise.resolve(formIFrame.onload)
 }
```

Comment:
This whole file is not really functional, could we create functions more explicit ?